src/interceptors/ClientRequest/utils/recordRawHeaders.test.ts

```diff
@@ -0,0 +1,130 @@
+  expect(getRawFetchHeaders(new Response().headers)).toEqual([])
+})
+
+it('records raw headers (Headers / object as init)', () => {
+  recordRawFetchHeaders()
+  const headers = new Headers({
+    'Content-Type': 'application/json',
+    'X-My-Header': '1',
+  })
+
+  expect(getRawFetchHeaders(headers)).toEqual([
+    ['Content-Type', 'application/json'],
+    ['X-My-Header', '1'],
+  ])
+  expect(Object.fromEntries(headers)).toEqual({
+    'content-type': 'application/json',
+    'x-my-header': '1',
+  })
+})
+
+it('records raw headers (Headers / array as init)', () => {
+  recordRawFetchHeaders()
+  const headers = new Headers([['X-My-Header', '1']])
+
+  expect(getRawFetchHeaders(headers)).toEqual([['X-My-Header', '1']])
+  expect(Object.fromEntries(headers)).toEqual({
+    'x-my-header': '1',
+  })
+})
+
+it('records raw headers (Headers / Headers as init', () => {
+  recordRawFetchHeaders()
+  const headers = new Headers([['X-My-Header', '1']])
+
+  expect(getRawFetchHeaders(new Headers(headers))).toEqual([
+    ['X-My-Header', '1'],
+  ])
+})
+
+it('records raw headers (Request / object as init)', () => {
+  recordRawFetchHeaders()
+  const request = new Request(url, {
+    headers: {
+      'Content-Type': 'application/json',
+      'X-My-Header': '1',
+    },
+  })
+
+  expect(getRawFetchHeaders(request.headers)).toEqual([
+    ['Content-Type', 'application/json'],
+    ['X-My-Header', '1'],
+  ])
+})
+
+it('records raw headers (Request / array as init)', () => {
+  recordRawFetchHeaders()
+  const request = new Request(url, {
+    headers: [['X-My-Header', '1']],
+  })
+
+  expect(getRawFetchHeaders(request.headers)).toEqual([['X-My-Header', '1']])
+})
+
+it('records raw headers (Request / Headers as init)', () => {
+  recordRawFetchHeaders()
+  const headers = new Headers([['X-My-Header', '1']])
+  const request = new Request(url, { headers })
+
+  expect(getRawFetchHeaders(request.headers)).toEqual([['X-My-Header', '1']])
+})
+
+it('records raw headers (Response / object as init)', () => {
+  recordRawFetchHeaders()
+  const response = new Response(null, {
+    headers: {
+      'Content-Type': 'application/json',
+      'X-My-Header': '1',
+    },
+  })
+
+  expect(getRawFetchHeaders(response.headers)).toEqual([
+    ['Content-Type', 'application/json'],
+    ['X-My-Header', '1'],
+  ])
+})
+
+it('records raw headers (Response / array as init)', () => {
+  recordRawFetchHeaders()
+  const response = new Response(null, {
+    headers: [['X-My-Header', '1']],
+  })
+
+  expect(getRawFetchHeaders(response.headers)).toEqual([['X-My-Header', '1']])
+})
+
+it('records raw headers (Response / Headers as init)', () => {
+  recordRawFetchHeaders()
+  const headers = new Headers([['X-My-Header', '1']])
+  const response = new Response(null, { headers })
+
+  expect(getRawFetchHeaders(response.headers)).toEqual([['X-My-Header', '1']])
+})
+
+it('stops recording once the patches are restored', () => {
```

Comment:
I'm wondering if we need the test I wrote if we have this test

Reply:
Unit vs integration. Let's keep yours. It tests that ClientRequest interceptors applies/restores the headers patching. The unit tests don't test that.